File: examples/currency/history/node/historicRatesForACurrency.js
import * as currency from "m3o/currency";

// Returns the historic rates for a currency on a given date
async function HistoricRatesForAcurrency() {
  let currencyService = new currency.CurrencyService(
    process.env.MICRO_API_TOKEN
  );
  let rsp = await currencyService.history({
    code: "USD",
    date: "2021-05-30",
  });
  console.log(rsp);
}

await HistoricRatesForAcurrency();

File: examples/go/currency/history/historicRatesForACurrency.go
package example

import (
	"fmt"
	"os"

	"github.com/micro/services/clients/go/currency"
)

// Returns the historic rates for a currency on a given date
func HistoricRatesForAcurrency() {
	currencyService := currency.NewCurrencyService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := currencyService.History(&currency.HistoryRequest{
		Code: "USD",
		Date: "2021-05-30",
	})
	fmt.Println(rsp, err)
}
